src/MonitorProfiler/Utilities/TypeNameUtilities.cpp

```diff
@@ -196,7 +199,20 @@ HRESULT TypeNameUtilities::GetTypeDefName(NameCache& nameCache, ModuleID moduleI
         {
             IfFailRet(pMDImport->GetNestedClassProps(tokenToProcess, &outerTokenType));
         }
-        nameCache.AddTokenData(moduleId, tokenToProcess, outerTokenType, tstring(wName));
+        else
+        {
+            std::string::size_type found = tstring(wName).find_last_of(L'.');
+
+            if (found != std::string::npos)
+            {
+                wcsncpy_s(wNamespace, wName, found);
```

Comment:
This area seems to be causing build failures on non-windows platforms - I'm not sure if this is the approach we'll continue using for getting the namespace (is there something cleaner than just parsing for the `.`?)

Reply:
Because you are ultimately passing `tstring` values to `AddTokenData`, just use the methods on `tstring` such as [substr](https://en.cppreference.com/w/cpp/string/basic_string/substr).